### DISCUSSION TITLE
What is Best Practice for Connecting Pre&Post Middleware?

### DISCUSSION THREAD

**Discussion by ajoslin103** [adonisjs/core#1665] [Help]
I wonder what is the best practice for connecting pre & post middlewares?

Having read this thread: https://forum.adonisjs.com/t/how-to-initiate-new-class-in-service-provider/6114

I like the look of the pattern @McSneaky provides:

1. Request comes in
2. Upstream middleware logs in
3. Request is processed
4. Downstream middleware logs out
5. Data is sent back to client

What is best practice for passing the connection gained in Step 2 on to Steps 3 & 4

?

**Comment by McSneaky** [comment]
Really late to party, but I'd set it to something in request. Like `request.user` or `request.costumLoginData` and then use that in next parts

**Reply by ajoslin103** [reply]
No worries, I didn't know if it was cool to add something to the request -- figured that was where it would go if A didn't have some smoother way about it.

I'm doing some ampq stuff and it can be a loooooong time between calls -- so I was thinking about connecting/disconnecting per call

So far I've been connecting when Provider is instantiated and it's been working ok

Thanks!
Al;
